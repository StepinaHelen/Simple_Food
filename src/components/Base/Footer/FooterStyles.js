import styled from "styled-components";

export const FooterContainer = styled.footer`
  background-color: #6b068a;
  color: white;
  width: 100%;
  display: block;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 19px;
  margin: 0 auto;

  p {
    padding: 40px 0;
    text-align: center;
    margin: 0px;
  }

  span {
    color: red;
  }
`;
