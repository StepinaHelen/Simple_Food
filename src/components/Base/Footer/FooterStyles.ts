import styled from "styled-components";

export const FooterContainer = styled.footer`
  background-color: ${({ theme }) => theme.colors.primary};
  color: ${({ theme }) => theme.colors.consantColor};
  width: 100%;
  display: block;
`;

export const Content = styled.div`
  ${({ theme }) => theme.fontText({})}
  margin: 0 auto;

  p {
    padding: 40px 0;
    text-align: center;
    margin: 0px;
  }

  span {
    color: ${({ theme }) => theme.colors.error};
  }
  a {
    color: ${({ theme }) => theme.colors.consantColor};
    display: inline-block;
    text-decoration: underline;
  }
`;
