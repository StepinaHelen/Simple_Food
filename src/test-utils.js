import { render } from "@testing-library/react";
import { QueryClientProvider, QueryClient } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { baseTheme } from "./styles/theme";
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={baseTheme}>
      <QueryClientProvider contextSharing={true} client={queryClient}>
        <BrowserRouter>{children}</BrowserRouter>
      </QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
