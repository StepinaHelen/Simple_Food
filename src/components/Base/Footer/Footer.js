import CommonContainer from "../Containers/CommonContainer";
import { FooterContainer, Content } from "./FooterStyles";

const Footer = () => {
  return (
    <FooterContainer>
      <CommonContainer>
        <Content>
          <p>
            &#169; 2022 | All Rights Reserved | Developed with &nbsp;
            <span>&#10084;</span>&nbsp; by Stepina Helen
          </p>
        </Content>
      </CommonContainer>
    </FooterContainer>
  );
};

export default Footer;
