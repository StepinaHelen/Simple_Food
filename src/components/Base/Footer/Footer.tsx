import CommonContainer from "components/Base/Containers/CommonContainer";
import { FooterContainer, Content } from "components/Base/Footer/FooterStyles";

const Footer = () => {
  return (
    <FooterContainer>
      <CommonContainer withMargin={false}>
        <Content>
          <p>
            &#169; 2022 | All Rights Reserved | Developed with &nbsp;
            <span>&#10084;</span>&nbsp; by &nbsp;
            <a href="https://www.linkedin.com/in/helen-stepina-a5b8b8211/">
              Stepina Helen
            </a>
          </p>
        </Content>
      </CommonContainer>
    </FooterContainer>
  );
};

export default Footer;
