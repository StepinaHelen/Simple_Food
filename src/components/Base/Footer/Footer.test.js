import { render, screen } from "test-utils";
import Footer from "./Footer";
describe("test Footer", () => {
  it("Footer renders", () => {
    render(<Footer />);
  });
  it("Footer snapshot", () => {
    const view = render(<Footer />);
    expect(view).toMatchSnapshot();
  });
  it("check Footer text", () => {
    render(<Footer />);
    expect(screen.getByText("Stepina Helen")).toBeInTheDocument();
  });
});
