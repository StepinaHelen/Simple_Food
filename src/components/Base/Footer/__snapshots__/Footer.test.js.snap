// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Footer Footer snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  padding-left: 15px;
  padding-right: 15px;
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  background-color: #6b068a;
  color: white;
  width: 100%;
  display: block;
}

.c2 {
  font-size: 16px;
  line-height: 19px;
  margin: 0 auto;
}

.c2 p {
  padding: 40px 0;
  text-align: center;
  margin: 0px;
}

.c2 span {
  color: red;
}

@media screen and (min-width:480px) {
  .c1 {
    width: 480px;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    width: 768px;
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    width: 1200px;
  }
}

<body>
    <div>
      <footer
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <p>
              © 2022 | All Rights Reserved | Developed with  
              <span>
                ❤
              </span>
                by
              <a
                href="https://www.linkedin.com/in/helen-stepina-a5b8b8211/"
              >
                Stepina Helen
              </a>
            </p>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": .c1 {
  padding-left: 15px;
  padding-right: 15px;
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  background-color: #6b068a;
  color: white;
  width: 100%;
  display: block;
}

.c2 {
  font-size: 16px;
  line-height: 19px;
  margin: 0 auto;
}

.c2 p {
  padding: 40px 0;
  text-align: center;
  margin: 0px;
}

.c2 span {
  color: red;
}

@media screen and (min-width:480px) {
  .c1 {
    width: 480px;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    width: 768px;
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    width: 1200px;
  }
}

<div>
    <footer
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <p>
            © 2022 | All Rights Reserved | Developed with  
            <span>
              ❤
            </span>
              by
            <a
              href="https://www.linkedin.com/in/helen-stepina-a5b8b8211/"
            >
              Stepina Helen
            </a>
          </p>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
